File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/SavedRequestCookieHelper.java
package uk.gov.justice.digital.hmpps.oauth2server.config;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import uk.gov.justice.digital.hmpps.oauth2server.utils.CookieHelper;

import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@Component
public class SavedRequestCookieHelper extends CookieHelper {

    @Autowired
    public SavedRequestCookieHelper(final SavedRequestCookieConfigurationProperties properties) {
        super(properties.getName(), properties.getExpiryTime());
    }

    void removeCookie(final HttpServletRequest request, final HttpServletResponse response) {
        final var removeSavedRequestCookie = new Cookie(getName(), "");
        removeSavedRequestCookie.setPath(request.getContextPath() + "/");
        removeSavedRequestCookie.setMaxAge(0);
        removeSavedRequestCookie.setSecure(request.isSecure());
        removeSavedRequestCookie.setHttpOnly(true);
        response.addCookie(removeSavedRequestCookie);
    }
}

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/SavedRequestCookieHelper.kt
package uk.gov.justice.digital.hmpps.oauth2server.config

import org.springframework.stereotype.Component
import uk.gov.justice.digital.hmpps.oauth2server.utils.CookieHelper
import javax.servlet.http.Cookie
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Component
class SavedRequestCookieHelper(properties: SavedRequestCookieConfigurationProperties) :
  CookieHelper(properties.name, properties.expiryTime) {

  fun removeCookie(request: HttpServletRequest, response: HttpServletResponse) {
    val removeSavedRequestCookie = Cookie(name, "")
    removeSavedRequestCookie.path = request.contextPath + "/"
    removeSavedRequestCookie.maxAge = 0
    removeSavedRequestCookie.secure = request.isSecure
    removeSavedRequestCookie.isHttpOnly = true
    response.addCookie(removeSavedRequestCookie)
  }
}
